src/store/modules/app.js

```diff
@@ -73,6 +84,46 @@ const getters = {
 };
 
 const actions = {
+  checkValidKeymap(_, keymap) {
+    if (
+      isUndefined(keymap.keyboard) ||
+      isUndefined(keymap.keymap) ||
+      isUndefined(keymap.layout) ||
+      isUndefined(keymap.layers)
+    ) {
+      return false;
+    }
+    return true;
+  },
+  validateKeyboard({ state }, keyboard) {
```

Comment:
why is this an action?

Reply:
getter is a property, action is a function. This does the validation, if the keyboard is in the availables keyboards.